sdk/src/main/java/io/customer/sdk/CustomerIO.kt

```diff
@@ -337,6 +344,14 @@ class CustomerIO internal constructor(
             val client = CustomerIO(diGraph)
             val logger = diGraph.logger
 
+            // cleanup of old reference if it exists
+            instance?.let {
+                // we need to unregister the previous activity lifecycle because it doesn't get garbage collected
+                // and will continue to hold a reference to the previous app context and return callbacks
+                appContext.unregisterActivityLifecycleCallbacks(it.diGraph.activityLifecycleCallbacks)
+                instance = null
```

Comment:
Why set instance to null? Can we remove this?

Reply:
What's wrong with setting the instance to `null`?  it will just make it available earlier for GC collection. Whats the downside of having it?

Reply:
It's a nit-picky thing that shouldn't block the PR. 

I didn't expect to see this line of code and didn't understand the reason for adding it (especially because the next line of code is `instance = client`), so I thought I would ask the purpose it serves. 

Since I don't see a reason to add it and it isn't really hurting anything, I don't have much else to add here?